File: frontend/src/types/status-effect.ts
export type StatusEffectType =
  | 'Buff'
  | 'Debuff'
  | 'Special'
  | 'Control'
  | 'Elemental'
  | 'Blessing'
  | 'Exclusive';

export interface StatusEffect {
  name: string;
  type: StatusEffectType;
  effect: string;
  remark: string;
}

File: backend/models/status_effect.py
from enum import Enum

from pydantic import BaseModel


class StatusEffectType(str, Enum):
    BUFF = "Buff"
    DEBUFF = "Debuff"
    SPECIAL = "Special"
    CONTROL = "Control"
    ELEMENTAL = "Elemental"
    BLESSING = "Blessing"
    EXCLUSIVE = "Exclusive"


class StatusEffect(BaseModel):
    name: str
    type: StatusEffectType
    effect: str
    remark: str
